Libraries/PermissionsAndroid/PermissionsAndroid.js

```diff
@@ -119,36 +113,25 @@ class PermissionsAndroid {
    * and then shows the system permission dialog
    */
   requestPermission(permission: string, rationale?: Rationale) : Promise<boolean> {
-    return new Promise((resolve, reject) => {
```

Comment:
The code got a bit messy here, how about,

```js
async requestPermission(permission: string, rationale?: Rationale) : Promise<boolean> {
  if (rationale) {
    const shouldShowRationale = await Permissions.shouldShowRequestPermissionRationale(permission);

    if (shouldShowRationale) {
      return new Promise((resolve, reject) => {
        DialogManagerAndroid.showAlert(
            rationale,
            () => {
              DialogManagerAndroid.showAlert({
                message: 'Error Requesting Permissions'
              }, {}, {});
              reject();
            },
            resolve(Permissions.requestPermission(permission));
          );
      });
    }
  }
  return Permissions.requestPermission(permission);
}
```

I just wrote the code here, didn't test it.